12-offer-encoding.md

```diff
@@ -0,0 +1,846 @@
+| 16   | Multi-part-payment support       | MPP/compulsory |
+| 17   | Multi-part-payment support       | MPP/optional   |
+
+The 'MPP support' invoice feature indicates that the payer MUST (16) or
+MAY (17) use multiple part payments to pay the invoice.
+
+Some implementations may not support MPP (e.g. for small payments), or
+may (due to capacity limits on a single channel) require it.
+
+## Requirements
+
+A writer of an invoice:
+  - MUST set `invoice_created_at` to the number of seconds since Midnight 1
+    January 1970, UTC when the offer was created.
+  - MUST set `invoice_amount` to the minimum amount it will accept, in units of 
+    the minimal lightning-payable unit (e.g. milli-satoshis for bitcoin) for
+    `invreq_chain`.
+  - if the invoice is in response to an `invoice_request`:
+    - MUST copy all non-signature fields from the `invoice_request` (including unknown fields).
+    - if `invreq_amount` is present:
+      - MUST set `invoice_amount` to `invreq_amount`
+    - otherwise:
+      - MUST set `invoice_amount` to the *expected amount*.
+  - otherwise (invoice not requested, e.g. for user to scan directly):
+    - MUST set `invreq_chain` as it would for an invoice_request.
+    - MUST set `offer_description` and `offer_node_id` as it would for an offer.
+    - MUST NOT set `invreq_payer_id`.
+  - MUST set `invoice_payment_hash` to the SHA256 hash of the
+    `payment_preimage` that will be given in return for payment.
+  - if `offer_node_id` is not present:
+    - if the `invoice_request` was not communicated securely (i.e. privately):
+      - SHOULD set `invoice_code` to a short string for verification.
+    - MUST NOT set `signature`.
+  - otherwise:
+    - MUST NOT set `invoice_code`.
+    - MUST specify exactly one signature TLV element: `signature`.
+      - MUST set `sig` to the signature using `offer_node_id` as described in [Signature Calculation](#signature-calculation).
+  - if it requires multiple parts to pay the invoice:
+    - MUST set `invoice_features`.`features` bit `MPP/compulsory`
+  - or if it allows multiple parts to pay the invoice:
+    - MUST set `invoice_features`.`features` bit `MPP/optional`
+  - if the expiry for accepting payment is not 7200 seconds after `invoice_created_at`:
+    - MUST set `invoice_relative_expiry`.`seconds_from_creation` to the number of
+      seconds after `invoice_created_at` that payment of this invoice should not be attempted.
+  - if it accepts onchain payments:
+    - MAY specify `invoice_fallbacks`
+    - MUST specify `invoice_fallbacks` in order of most-preferred to least-preferred
+      if it has a preference.
+    - for the bitcoin chain, it MUST set each `fallback_address` with
+      `version` as a valid witness version and `address` as a valid witness
+      program
+  - MUST include `invoice_paths` containing one or more paths to the node.
+    - MUST specify `invoice_paths` in order of most-preferred to least-preferred if it has a preference.
```

Comment:
Should the reader section indicate how paths should be chosen?

Reply:
Added a general admonition: 

```
  - SHOULD prefer to use earlier `invoice_paths` over later ones if it has no other reason for preference.
```